tsdb/head.go

```diff
@@ -2040,106 +2040,118 @@ func (h *Head) updateWALReplayStatusRead(current int) {
+	}
+	return &h
+}
+
 func GenerateTestGaugeHistograms(n int) (r []*histogram.Histogram) {
 	for x := 0; x < n; x++ {
-		i := rand.Intn(n)
-		r = append(r, &histogram.Histogram{
-			CounterResetHint: histogram.GaugeType,
-			Count:            10 + uint64(i*8),
-			ZeroCount:        2 + uint64(i),
-			ZeroThreshold:    0.001,
-			Sum:              18.4 * float64(i+1),
-			Schema:           1,
-			PositiveSpans: []histogram.Span{
-				{Offset: 0, Length: 2},
-				{Offset: 1, Length: 2},
-			},
-			PositiveBuckets: []int64{int64(i + 1), 1, -1, 0},
-			NegativeSpans: []histogram.Span{
-				{Offset: 0, Length: 2},
-				{Offset: 1, Length: 2},
-			},
-			NegativeBuckets: []int64{int64(i + 1), 1, -1, 0},
-		})
+		r = append(r, GenerateTestGaugeHistogram(rand.Intn(n)))
 	}
 	return r
 }
 
+func GenerateTestGaugeHistogram(i int) *histogram.Histogram {
+	return &histogram.Histogram{
+		CounterResetHint: histogram.GaugeType,
+		Count:            10 + uint64(i*8),
+		ZeroCount:        2 + uint64(i),
+		ZeroThreshold:    0.001,
+		Sum:              18.4 * float64(i+1),
+		Schema:           1,
+		PositiveSpans: []histogram.Span{
+			{Offset: 0, Length: 2},
+			{Offset: 1, Length: 2},
+		},
+		PositiveBuckets: []int64{int64(i + 1), 1, -1, 0},
+		NegativeSpans: []histogram.Span{
+			{Offset: 0, Length: 2},
+			{Offset: 1, Length: 2},
+		},
+		NegativeBuckets: []int64{int64(i + 1), 1, -1, 0},
+	}
+}
+
 func GenerateTestFloatHistograms(n int) (r []*histogram.FloatHistogram) {
 	for i := 0; i < n; i++ {
-		h := histogram.FloatHistogram{
-			Count:         10 + float64(i*8),
-			ZeroCount:     2 + float64(i),
-			ZeroThreshold: 0.001,
-			Sum:           18.4 * float64(i+1),
-			Schema:        1,
-			PositiveSpans: []histogram.Span{
-				{Offset: 0, Length: 2},
-				{Offset: 1, Length: 2},
-			},
-			PositiveBuckets: []float64{float64(i + 1), float64(i + 2), float64(i + 1), float64(i + 1)},
-			NegativeSpans: []histogram.Span{
-				{Offset: 0, Length: 2},
-				{Offset: 1, Length: 2},
-			},
-			NegativeBuckets: []float64{float64(i + 1), float64(i + 2), float64(i + 1), float64(i + 1)},
-		}
-		if i > 0 {
-			h.CounterResetHint = histogram.NotCounterReset
-		}
-		r = append(r, &h)
+		r = append(r, GenerateTestFloatHistogram(i))
 	}
-
 	return r
 }
 
+func GenerateTestFloatHistogram(i int) *histogram.FloatHistogram {
+	h := histogram.FloatHistogram{
+		Count:         10 + float64(i*8),
+		ZeroCount:     2 + float64(i),
+		ZeroThreshold: 0.001,
+		Sum:           18.4 * float64(i+1),
+		Schema:        1,
+		PositiveSpans: []histogram.Span{
+			{Offset: 0, Length: 2},
+			{Offset: 1, Length: 2},
+		},
+		PositiveBuckets: []float64{float64(i + 1), float64(i + 2), float64(i + 1), float64(i + 1)},
+		NegativeSpans: []histogram.Span{
+			{Offset: 0, Length: 2},
+			{Offset: 1, Length: 2},
+		},
+		NegativeBuckets: []float64{float64(i + 1), float64(i + 2), float64(i + 1), float64(i + 1)},
+	}
+	if i > 0 {
+		h.CounterResetHint = histogram.NotCounterReset
+	}
```

Comment:
Similar comment as `GenerateTestHistogram`. Alternatively, we can add a comment to these functions that they set NotCounterReset if i>0.

Reply:
I've added a comment about _not_ setting the counter reset hint .